src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1079,6 +1017,56 @@ impl<T: 'static> EventProcessor<T> {
                         }
                     }
 
+                    // The regular KeyPress event has a problem where if you press a dead key, a KeyPress
+                    // event won't be emitted. XInput 2 does not have this problem.
+                    ffi::XI_KeyPress | ffi::XI_KeyRelease => {
+                        if let Some(active_window) = self.active_window {
+                            let state = if xev.evtype == ffi::XI_KeyPress {
+                                Pressed
+                            } else {
+                                Released
+                            };
+
+                            let xkev: &ffi::XIDeviceEvent = unsafe { &*(xev.data as *const _) };
+
+                            // We use `self.active_window` here as `xkev.event` has a completely different
+                            // value for some reason.
+                            let window_id = mkwid(active_window);
+
+                            let device_id = mkdid(xkev.deviceid);
+                            let keycode = xkev.detail as u32;
+
+                            let keycode = keycode - KEYCODE_OFFSET as u32;
+                            let mut ker = self.kb_state.process_key_event(keycode, state);
```

Comment:
Is `ker` a typo for `key`?

Reply:
Nope, it's an abbreviation for the `KeyEventResults` struct, which `KbState::process_key_event` returns.